homeassistant/components/deconz/config_flow.py

```diff
@@ -217,13 +204,13 @@ def _update_entry(self, entry, host, port, api_key=None):
 
         This flow is triggered by the discovery component.
         """
-        bridgeid = user_input[CONF_SERIAL]
-        gateway_entries = configured_gateways(self.hass)
+        self.bridgeid = normalize_bridge_id(user_input[CONF_SERIAL])
+        await self.async_set_unique_id(self.bridgeid)
```

Comment:
This will abort if another flow is in progress with same unique ID. You probably don't want that. You can also wait with this until after you know the gateway doesn't exist.

Reply:
This is only hassio, there could only be one right?

Reply:
What if SSDP detects the Hass.io add-on before the hass.io discovery comes in ?

Reply:
Is ssdp discovery enabled for the addon? @frenck?

Reply:
It can be enabled.

TBH I think this is a question on a different level: What if a unique ID for discovery is done via multiple sources? Which gets the presidency?

I think this needs resolving on a different level.

Reply:
In this case there should be no reason to care about ssdp right?

Reply:
@balloob @frenck Ok like this?

Reply:
At this point, I would say this is OK!
I think we should let @balloob make the final call on this.